### ISSUE TITLE
Casting to Enumerable of implementing type of element fails with invalid cast

### ISSUE THREAD

**Issue by michaelcheers** [bridgedotnet/Bridge#3245]
Casting to Enumerable of implementing type of element fails with invalid cast.

### Steps To Reproduce

https://dev.deck.net/1ee8092d7b95d312a2bbe298d3fa9614/third

```csharp
using System;
using System.Collections.Generic;
using System.Linq;

public class A {}

public class B : A {}

public static class Program
{
	public static void Main(string[] args)
	{
		var listB = new List<B>
		{
			new B(),
			new B()
		};
		try
		{
			var bAsEnumerableA = (IEnumerable<A>)listB;
			Console.WriteLine(bAsEnumerableA.Count() == 2 ? "Passed" : "Failed");
		}
		catch
		{
			Console.WriteLine("Failed");
		}
	}
}
```

**Comment by vladsch** [comment]
The same issue https://github.com/bridgedotnet/Bridge/issues/3244
Fix for #3244 should affect on this also.

**Comment by fabriciomurta** [comment]
As @vladsch pointed, in ended up this duplicates issue #3244. Notwithstanding we added a test case based on your report to the tests of #3244 so we don't let this slip again! Thanks for the clean report!

That said, please follow #3244 for updates regarding this issue.
